internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -356,17 +359,19 @@ func (f *FleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	resp, took, err := cmd.Execute(ctx, req)
 	if isUnauth(err) {
 		f.unauthCounter++
-
-		if f.shouldUnenroll() {
-			f.log.Warnf("retrieved an invalid api key error '%d' times. Starting to unenroll the elastic agent.", f.unauthCounter)
-			return &fleetapi.CheckinResponse{
-				Actions: []fleetapi.Action{&fleetapi.ActionUnenroll{ActionID: "", ActionType: "UNENROLL", IsDetected: true}},
-			}, took, nil
+		if f.shouldUseLongSched() {
+			f.log.Warnf("retrieved an invalid api key error '%d' times. will use long scheduler", f.unauthCounter)
+			f.scheduler = scheduler.NewPeriodic(defaultGatewaySettings.ErrDuration)
```

Comment:
This should still be using jitter, just on a longer request duration. You do not want 50,000 agents accidentally checking it at the exact same moment unnecessarily.

Reply:
Possibly it is easier to adjust the duration up and down than to replace the whole scheduler.